### ISSUE TITLE
Repro's handling of Timer C

### ISSUE THREAD

**Issue by IvanRibakov** [resiprocate/resiprocate#209]
As far as I can tell, current Repro behaviour when Timer C is fired is to cancel ALL client transactions:

```
// repro/ResponseContext.cxx
void
ResponseContext::processTimerC()
{
   if (!mRequestContext.mHaveSentFinalResponse)
   {
      InfoLog(<<"Canceling client transactions due to timer C.");
      cancelAllClientTransactions();
   }
}
```

```
INFO | 20220118-081343.009 | repro | REPRO:APP | 139697393129024 | RequestContext.cxx:895 | Updating timer C.
INFO | 20220118-081343.009 | repro | REPRO:APP | 139697393129024 | Proxy.cxx:569 | Posting timer C (8sec)
INFO | 20220118-081350.641 | repro | REPRO:APP | 139697393129024 | RequestContext.cxx:660 | RequestContext::process(ApplicationMessage) TimerCMessage(tid=-811889-1-0)
INFO | 20220118-081350.857 | repro | REPRO:APP | 139697393129024 | RequestContext.cxx:660 | RequestContext::process(ApplicationMessage) TimerCMessage(tid=-811889-1-0)
INFO | 20220118-081351.009 | repro | REPRO:APP | 139697393129024 | RequestContext.cxx:660 | RequestContext::process(ApplicationMessage) TimerCMessage(tid=-811889-1-0)
INFO | 20220118-081351.009 | repro | REPRO:APP | 139697393129024 | ResponseContext.cxx:981 | Canceling client transactions due to timer C.
INFO | 20220118-081351.009 | repro | REPRO:APP | 139697393129024 | ResponseContext.cxx:272 | Cancel ALL client transactions: 2 pending, 1 active.
```

Unless I'm misunderstanding something, that behaviour does not seem to be aligned with [RFC3261 which says following](https://datatracker.ietf.org/doc/html/rfc3261#section-16.8):

> 16.8 Processing Timer C
>
>
>   If timer C should fire, the proxy MUST either reset the timer with
>   any value it chooses, or terminate the client **transaction**.

Notice it says `terminate the client transaction` (singular).

Reason I'm bringing this up is because I'm trying to configure Repro to behave in a way that after a client transaction is terminated due to Timer C, next target in the delivery list is used to route the call to.

So, to summarise:
1. Is current Repro handling of Timer C valid?
2. Is it possible to continue routing call to other targets after one of them triggered Timer C?

**Comment by TheQue42** [comment]
It definitely seems wrong to use a single timer c for all txns, since 1xx will/should restart the timer, and that means that a 180 response on one transaction, can keep all transactions going.

**Comment by IvanRibakov** [comment]
Answering one of my earlier questions:

> 2. Is it possible to continue routing call to other targets after one of them triggered Timer C?

Probably not after TimerC expiry, but a similar behaviour can be achieved if using Q-Value Target Handler Baboon by adjusting following configuration parameters:
- `QValueCancelBetweenForkGroups = true`
- `QValueMsBeforeCancel = ...`
